Sources/OktaOAuth2/Authentication/JWTAuthorizationFlow.swift

```diff
@@ -0,0 +1,151 @@
+//
+// Copyright (c) 2024-Present, Okta, Inc. and/or its affiliates. All rights reserved.
+// The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+//
+// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and limitations under the License.
+//
+
+import Foundation
+import AuthFoundation
+
+/// An authentication flow class that implements the JWT Authorization Bearer Flow, for authenticating users using JWTs signed by a trusted key.
+public class JWTAuthorizationFlow: AuthenticationFlow {
+    /// The OAuth2Client this authentication flow will use.
+    public let client: OAuth2Client
+    
+    /// Indicates whether or not this flow is currently in the process of authenticating a user.
+    /// ``JWTAuthorizationFlow/init(issuer:clientId:scopes:)``
+    public private(set) var isAuthenticating: Bool = false {
+        didSet {
+            guard oldValue != isAuthenticating else {
+                return
+            }
+            
+            if isAuthenticating {
+                delegateCollection.invoke { $0.authenticationStarted(flow: self) }
+            } else {
+                delegateCollection.invoke { $0.authenticationFinished(flow: self) }
+            }
```

Comment:
Do you want this to call inline or should it be dispatched? Maybe consider changing to a function instead of a side effect of a variable change?

Reply:
The SDK affords developers the option to interact with the APIs using either completion blocks or async results, as well as responding to delegate calls.  So I've tried to centralize the notification logic as much as possible to ensure consistency.

I agree with your point though. I have some in-flight work that I haven't completed, but it introduces support for dispatch queues to be used in delegate calls, as well as propagates task cancellation handles up through the SDK, to allow async operations to be cancelled. That involves quite a bit of concurrency testing, so while the work is done, I wasn't happy with the level of test coverage, so I've been holding off on it.

Since I'll be pushing out a major release sometime soon, I might take the opportunity to finish that work.